stellar-contract-env-common/src/env.rs

```diff
@@ -86,6 +86,8 @@ macro_rules! call_macro_with_all_host_functions {
                 {"$3", fn map_len(m:Object) -> RawVal}
                 {"$4", fn map_keys(m:Object) -> Object}
                 {"$5", fn map_has(m:Object,k:RawVal) -> RawVal}
+                {"$6", fn map_lower_bound(m:Object,k:RawVal) -> RawVal}
```

Comment:
❓ Seeing the function definition here prompts some questions about the map type and CAP-15:
- How will an ScoMap with multiple entries of the same key be treated?
   - Will core consider such a map invalid and fail any transaction that attempts to input such a map? 
- In what order will the map be iterated by these functions, will it be:
   - Undefined or bucket order and essentially unpredictable to the user, although consistent if repeated for a map not altered?
   - Random order so as to make the unpredictable nature of the order clearer and precent reliance on happenstance ordering? (e.g. like Go)
   - Ordered by the keys based on the keys ordering?
      - If it is ordered by the keys how will keys of different types be ordered?
   - Ordered based on insertion, which is convenient in many use cases? (e.g. Ruby, JavaScript)

cc @graydon